internal/runbits/buildscript/buildscript.go

```diff
@@ -0,0 +1,86 @@
+package buildscript
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	bpModel "github.com/ActiveState/cli/pkg/platform/api/graphql/model/buildplanner"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildscript"
+	"github.com/ActiveState/cli/pkg/platform/runtime/model"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+func getRemoteBuildScript(proj *project.Project, customCommit *strfmt.UUID, auth *authentication.Auth) (*bpModel.BuildScript, error) {
+	bp := model.NewBuildPlanner(auth)
+	commitID := proj.CommitUUID()
+	if customCommit != nil {
+		commitID = *customCommit
+	}
+	return bp.GetBuildScript(proj.Owner(), proj.Name(), commitID.String())
+}
+
+// Sync synchronizes the local build script with the remote one.
+// If a commit ID is given, a local mutation has occurred (e.g. added a package, pulled, etc.), so
+// pull in the new build script. Otherwise, if there are local build script changes, create a new
+// commit with them in order to update the remote one.
+func Sync(proj *project.Project, commitID *strfmt.UUID, out output.Outputer, auth *authentication.Auth) error {
+	file, err := buildscript.Get(proj.Dir())
+	if err != nil && !buildscript.IsDoesNotExistError(err) {
+		return errs.Wrap(err, "Could not get local build script")
+	}
+
+	script, err := getRemoteBuildScript(proj, commitID, auth)
+	if err != nil {
+		return errs.Wrap(err, "Could not get remote build script")
+	}
+
+	if file != nil {
+		if file.Script.Equals(script) {
+			return nil // nothing to do
+		}
+		// Note: merging and/or conflict resolution will happen in another ticket.
+		// For now, if commitID is given, a mutation happened, so prefer the remote build script.
+		// Otherwise, prefer local changes.
+		if commitID == nil {
+			script = file.Script
+		}
+
+		out.Notice(locale.Tl("buildscript_update", "Updating project to reflect build script changes..."))
+
+		bp := model.NewBuildPlanner(auth)
+		// Note: these commits will be staged commits in a later ticket.
+		commit, err := bp.PushCommit(model.PushCommitParams{
+			Owner:        proj.Owner(),
+			Project:      proj.Name(),
+			ParentCommit: proj.CommitID(),
+			BranchRef:    proj.BranchName(),
+			Description:  locale.Tl("buildscript_commit_description", "Update project due to build script change."),
+			Script:       script,
+		})
+		if err != nil {
+			return errs.Wrap(err, "Could not update project to reflect build script changes.")
+		}
+		newCommitID := strfmt.UUID(commit.CommitID)
+		commitID = &newCommitID
+
+		script, err = getRemoteBuildScript(proj, commitID, auth)
```

Comment:
This may not be necessary if the platform doesn't change the given script?

Reply:
No, it shouldn't be. Can you verify that the platform won't change the script without making the request though? This will require more than just checking the requirements. We'll neet to ensure the timestamp doesn't change as well.

Reply:
I cannot verify that, and even so, there's no guarantee that it will not change in the future. Perhaps it's best to do this "just in case".